### PR TITLE
[OCCM] Improve documentation

### PR DESCRIPTION
**Pull request by lingxiankong**
**The binaries affected**:

- [x] openstack-cloud-controller-manager
- [ ] cinder-csi-plugin
- [ ] k8s-keystone-auth
- [ ] client-keystone-auth
- [ ] octavia-ingress-controller
- [ ] manila-csi-plugin
- [ ] manila-provisioner
- [ ] magnum-auto-healer
- [ ] barbican-kms-plugin

**What this PR does / why we need it**:
- Merge several files for openstack-cloud-controller-manager to make it easy to read and find useful information in a single place.
- Add ToC in the markdown file.

**Which issue this PR fixes**:
fixes #

**Special notes for reviewers**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: docs/provider-configuration.md [hunk 1]
```diff
@@ -179,8 +179,21 @@ These configuration options for the OpenStack provider pertain to the load
 balancer and should appear in the `[LoadBalancer]` section of the `$CLOUD_CONFIG`
 file.
 
+Although the openstack-cloud-controller-manager was initially implemented with
+Neutron-LBaaS support, Octavia is recommended now because Neutron-LBaaS has
+been deprecated since Queens OpenStack release cycle and not longer accepted
```
Comment (ramineni) [inline docs/provider-configuration.md]: s/not longer expected/no longer accepts

File: docs/provider-configuration.md [hunk 2]
```diff
@@ -179,8 +179,21 @@ These configuration options for the OpenStack provider pertain to the load
...
+based on Octavia enabled OpenStack environment. Functionalities are not
+guaranteed if using Neutron-LBaaS.
+
 ##### Load Balancer Optional Parameters
 
+* `use-octavia`: Used to determine whether to look for and use an
+  Octavia LBaaS V2 service catalog endpoint. Valid values are `true` or `false`.
+  Where `true` is specified and an Octavia LBaaS V2 entry can not be found, the
+  provider will fall back and attempt to find a Neutron LBaaS V2 endpoint
+  instead. The default value is `false`.
```
Comment (ramineni) [inline docs/provider-configuration.md]: @lingxiankong so by default code uses neutron Lbaas? shouldnt we change to default to octavia as the prior one is deprecated?

Comment (lingxiankong) [reply docs/provider-configuration.md]: Yeah, sounds good to me, we are still getting some that related to Neutron-LBaaS which is annoying. I will submit a patchset with that information.

Comment (jichenjc) [reply docs/provider-configuration.md]: +1 to default to Octavia
and just to confirm (not sure ) we don't have anything that only on neutron-lbaas but not on octavia, correct?

Comment (lingxiankong) [reply docs/provider-configuration.md]: No, we don't

Comment (jichenjc) [reply docs/provider-configuration.md]: ok, +1 to me ,thanks

**Comment by lingxiankong** [comment]
@ramineni @chrigl

**Comment by chrigl** [comment]
@lingxiankong I guess, defaulting to Octavia will surprise people. Should we maybe highlight this in README.md?

**Comment by lingxiankong** [comment]
> @lingxiankong I guess, defaulting to Octavia will surprise people. Should we maybe highlight this in README.md?

Sure, I will add some description in `using-openstack-cloud-controller-manager.md`

**Comment by lingxiankong** [comment]
/override openlab/cloud-provider-openstack-acceptance-test-csi-manila

This job needs to be fixed in openlab as part of https://github.com/kubernetes/cloud-provider-openstack/issues/883

File: pkg/cloudprovider/providers/openstack/openstack.go [hunk 3]
```diff
@@ -365,6 +365,11 @@ func ReadConfig(config io.Reader) (Config, error) {
 		return Config{}, fmt.Errorf("no OpenStack cloud provider config file given")
 	}
 	var cfg Config
+
+	// Set default values
+	cfg.LoadBalancer.UseOctavia = true
+	cfg.LoadBalancer.InternalLB = false
```
Comment (jichenjc) [inline pkg/cloudprovider/providers/openstack/openstack.go]: I understand this is improve doc so I think we might want to separate those 2 
one is to set default value (code change) and one is overall doc update.

Comment (lingxiankong) [reply pkg/cloudprovider/providers/openstack/openstack.go]: This is part of the intentional change, as we said in doc we will support octavia by default.

Comment (jichenjc) [reply pkg/cloudprovider/providers/openstack/openstack.go]: ok, no strong opinion , but usually refactory the doc and do function changes should be separated

Comment (ramineni) [reply pkg/cloudprovider/providers/openstack/openstack.go]: @lingxiankong As its breaking change for existing users of Neutron Lbaas, IMO, separate PR would be good , where we can add release note mentioning the same, Add PR description/title accordingly, would be easy to track which PR introduced the change.
And also, if they still want to use, they just need to change the flag to false right , and we can remove the full support after 2 releases or so once we update the documentation on deprecation. Thoughts?

Comment (lingxiankong) [reply pkg/cloudprovider/providers/openstack/openstack.go]: ok, thanks @jichenjc @ramineni

**Comment by lingxiankong** [comment]
/approve

Will submit another PR to make use-octavia=true the default option.

**Pull request merged**